### PR TITLE
Add new package: davix

### PR DESCRIPTION
**Pull request by JavierCVilla**

### CODE REVIEW

File: var/spack/repos/builtin/packages/davix/package.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    list_url = "http://grid-deployment.web.cern.ch/grid-deployment/dms/lcgutil/tar/davix/"
+
+    version('0.6.7',     'f811647d924a6dc5852c92110359ed91')
+
+    depends_on('libxml2')
+    depends_on('libuuid')
+    depends_on('pkgconfig')
+    depends_on('openssl')
+
+    patch('davix-0.6.7-uuid.patch', when="@0.6.7")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/davix/package.py]: Has this patch been submitted upstream?

File: var/spack/repos/builtin/packages/davix/package.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+
+class Davix(CMakePackage):
+    """High-performance file management over WebDAV/HTTP."""
+
+    homepage = "https://dmc.web.cern.ch/projects/davix"
+    url      = "http://grid-deployment.web.cern.ch/grid-deployment/dms/lcgutil/tar/davix/0.6.7/davix-0.6.7.tar.gz"
+    list_url = "http://grid-deployment.web.cern.ch/grid-deployment/dms/lcgutil/tar/davix/"
+
+    version('0.6.7',     'f811647d924a6dc5852c92110359ed91')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/davix/package.py]: Can you remove the extra spaces here?

File: var/spack/repos/builtin/packages/davix/package.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+from spack import *
+
+
+class Davix(CMakePackage):
+    """High-performance file management over WebDAV/HTTP."""
+
+    homepage = "https://dmc.web.cern.ch/projects/davix"
+    url      = "http://grid-deployment.web.cern.ch/grid-deployment/dms/lcgutil/tar/davix/0.6.7/davix-0.6.7.tar.gz"
+    list_url = "http://grid-deployment.web.cern.ch/grid-deployment/dms/lcgutil/tar/davix/"
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/davix/package.py]: You might need to add `list_depth = 1`. You can run `spack versions` to test.

File: var/spack/repos/builtin/packages/davix/package.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    version('0.6.7',     'f811647d924a6dc5852c92110359ed91')
+
+    depends_on('libxml2')
+    depends_on('libuuid')
+    depends_on('pkgconfig')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/davix/package.py]: This is usually `type='build'`.

File: var/spack/repos/builtin/packages/davix/package.py (hunk 1 above)
Comment (JavierCVilla) [reply var/spack/repos/builtin/packages/davix/package.py]: Yes, it has.

File: var/spack/repos/builtin/packages/davix/package.py (hunk 3 above)
Comment (JavierCVilla) [reply var/spack/repos/builtin/packages/davix/package.py]: You're right, using `list_depth` it finds all the available versions.

**Pull request merged**